# Student Perfomance Factors

https://www.kaggle.com/datasets/lainguyn123/student-performance-factors

In [46]:
import pandas as pd

In [78]:
df = pd.read_csv("/content/drive/MyDrive/Elogroup/CNH data literacy/Beg/M6/StudentPerformanceFactors.csv")

In [79]:
df.head()

,Hours_Studied,Attendance,Parental_Involvement,Access_to_Resources,Extracurricular_Activities,Sleep_Hours,Previous_Scores,Motivation_Level,Internet_Access,Tutoring_Sessions,Family_Income,Teacher_Quality,School_Type,Peer_Influence,Physical_Activity,Learning_Disabilities,Parental_Education_Level,Distance_from_Home,Gender,Exam_Score
0,23,84,Low,High,No,7,73,Low,Yes,0,Low,Medium,Public,Positive,3,No,High School,Near,Male,67
1,19,64,Low,Medium,No,8,59,Low,Yes,2,Medium,Medium,Public,Negative,4,No,College,Moderate,Female,61
2,24,98,Medium,Medium,Yes,7,91,Medium,Yes,2,Medium,Medium,Public,Neutral,4,No,Postgraduate,Near,Male,74
3,29,89,Low,Medium,Yes,8,98,Medium,Yes,1,Medium,Medium,Public,Negative,4,No,High School,Moderate,Male,71
4,19,92,Medium,Medium,Yes,6,65,Medium,Yes,3,Medium,High,Public,Neutral,4,No,College,Near,Female,70


In [80]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6607 entries, 0 to 6606
Data columns (total 20 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Hours_Studied               6607 non-null   int64 
 1   Attendance                  6607 non-null   int64 
 2   Parental_Involvement        6607 non-null   object
 3   Access_to_Resources         6607 non-null   object
 4   Extracurricular_Activities  6607 non-null   object
 5   Sleep_Hours                 6607 non-null   int64 
 6   Previous_Scores             6607 non-null   int64 
 7   Motivation_Level            6607 non-null   object
 8   Internet_Access             6607 non-null   object
 9   Tutoring_Sessions           6607 non-null   int64 
 10  Family_Income               6607 non-null   object
 11  Teacher_Quality             6529 non-null   object
 12  School_Type                 6607 non-null   object
 13  Peer_Influence              6607 non-null   obje

There are some null values here. For the purpose of the activity, let's remove the lines in which they occur.

In [81]:
df.dropna(inplace=True)

In [82]:
for col in df.columns:
  print(col)
  print(df[col].value_counts())

Hours_Studied
Hours_Studied
20    448
19    425
21    420
18    393
23    389
22    386
17    370
24    343
16    337
15    303
25    281
14    257
26    255
27    217
13    213
12    189
28    169
11    140
29    131
30    119
10     91
9      79
31     75
8      55
32     54
7      51
33     37
34     27
5      21
35     20
6      17
4      16
3      12
36     10
38      7
39      6
37      6
2       4
1       3
43      1
44      1
Name: count, dtype: int64
Attendance
Attendance
67     182
76     179
98     178
77     177
64     176
94     175
91     169
84     169
79     169
69     166
68     166
82     165
96     164
80     163
93     163
73     162
81     161
72     160
95     160
74     159
78     158
89     157
61     155
63     153
88     153
97     153
83     152
65     152
90     151
92     150
99     150
71     150
86     149
70     148
87     148
62     146
75     144
85     142
66     140
60      85
100     79
Name: count, dtype: int64
Parental_Involvement
Parental_Involve

There are no values like "Unknown" or mispells that need fixing.

## Descriptive Statistics with `describe()`

The first step in a statistical analysis is to understand the general structure of the numerical variables in the dataset.

The method `describe()` gives a statistical summary of the numerical columns. It helps us quickly identify the central tendency, dispersion, and range of each variable.

The main statistics returned are:

- `count`: number of non-missing values
- `mean`: average value
- `std`: standard deviation, which measures how spread out the values are
- `min`: smallest value
- `25%`: first quartile, where 25% of the data is below this value
- `50%`: median, where 50% of the data is below this value
- `75%`: third quartile, where 75% of the data is below this value
- `max`: largest value

This summary is useful to detect variables with very different scales, possible outliers, missing values, and general patterns in the dataset.

In [83]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
Hours_Studied,6378.0,19.977109,5.985460,1.0,16.0,20.0,24.0,44.0
Attendance,6378.0,80.020853,11.550723,60.0,70.0,80.0,90.0,100.0
Sleep_Hours,6378.0,7.034964,1.468033,4.0,6.0,7.0,8.0,10.0
Previous_Scores,6378.0,75.066165,14.400389,50.0,63.0,75.0,88.0,100.0
Tutoring_Sessions,6378.0,1.495296,1.233984,0.0,1.0,1.0,2.0,8.0
Physical_Activity,6378.0,2.972719,1.028926,0.0,2.0,3.0,4.0,6.0
Exam_Score,6378.0,67.252117,3.914217,55.0,65.0,67.0,69.0,101.0


### Hours_Studied

- Students studied an average of approximately 20 hours, with a standard deviation of about 6 hours. This means that there is a high variation of study time.

- The minimum value is 1 hour, while the maximum is 44 hours. The maximum is much higher than the third quartile, which is 24 hours, so some students studied far more than most of the group.

- The median is 20 hours, very close to the mean, suggesting that the distribution is reasonably centered.

### Attendance

- The average attendance is approximately 80%, with values ranging from 60% to 100%.

- The quartiles show a balanced distribution: 25% of students have attendance below 70%, 50% below 80%, and 75% below 90%. This suggests that attendance is well spread across the range.

- Since the minimum is 60%, there are no extremely low attendance values in this dataset.

### Sleep_Hours

- Students sleep an average of about 7 hours per night, which is also the median value.

- The standard deviation is relatively small, around 1.47 hours, meaning that most students have sleep durations close to the average.

- The values range from 4 to 10 hours. These limits seem realistic, although students sleeping only 4 hours may represent a group with very low sleep.

### Previous_Scores

- The average previous score is approximately 75, with a median also equal to 75.

- The values range from 50 to 100, showing that the dataset includes students with both lower and higher previous academic performance.

- The standard deviation is about 14.4, which indicates a considerable variation in previous scores.

### Tutoring_Sessions

- Students attended an average of about 1.5 tutoring sessions.

- The median is 1, and 75% of students attended up to 2 sessions. This suggests that most students had few tutoring sessions.

- However, the maximum is 8, which is much higher than the third quartile. This may indicate that a small group of students got more tutoring support.

### Physical_Activity

- Students practiced physical activity around 3 times on average.

- The median is also 3, and the interquartile range goes from 2 to 4. This suggests that most students have a moderate level of physical activity.

- The values range from 0 to 6, showing that the dataset includes both inactive students and students with frequent physical activity.

### Exam_Score

- The average exam score is approximately 67.25, with a median of 67. The mean and median are very close, suggesting that most exam scores are concentrated around the center.

- The standard deviation is low, around 3.91, which means that exam scores do not vary as much as some other variables.

- The minimum score is 55, while the maximum is 101. Since exam scores usually have a maximum of 100, the value 101 should be investigated. It may be an error, an extra-credit score, or a special case in the dataset.

The method `describe()` can also be used to summarize categorical columns.


It returns different information:

- `count`: number of non-missing values
- `unique`: number of different categories
- `top`: most frequent category
- `freq`: frequency of the most frequent category

Using `include="all"` allows us to analyze both numerical and categorical variables in the same summary table.

In [84]:
df.describe(include="object").T

,count,unique,top,freq
Parental_Involvement,6378,3,Medium,3251
Access_to_Resources,6378,3,Medium,3204
Extracurricular_Activities,6378,2,Yes,3807
Motivation_Level,6378,3,Medium,3237
Internet_Access,6378,2,Yes,5893
Family_Income,6378,3,Low,2582
Teacher_Quality,6378,3,Medium,3826
School_Type,6378,2,Public,4434
Peer_Influence,6378,3,Positive,2553
Learning_Disabilities,6378,2,No,5710


## Quartiles, Percentiles, IQR and Boxplot

Before looking for unusual values, we need to understand how the data is distributed.

A **percentile** indicates the value below which a certain percentage of the data falls.

For example:

- The 25th percentile means that 25% of the values are below that point.
- The 50th percentile means that 50% of the values are below that point.
- The 75th percentile means that 75% of the values are below that point.
- The 99th percentile means that 99% of the values are below that point.

In class environments, this is talked about a lot: "the student was in the 90th percentile", meaning it was better than 90% of the class.

Quartiles are specific percentiles that divide the data into four parts:

- `Q1`: first quartile, equivalent to the 25th percentile
- `Q2`: second quartile, equivalent to the 50th percentile or median
- `Q3`: third quartile, equivalent to the 75th percentile

The **IQR**, or interquartile range, measures the spread of the middle 50% of the data.

It is calculated as:

\[
IQR = Q3 - Q1
\]

The IQR is useful because it focuses on the central part of the distribution and is less affected by extremely high or extremely low values.

In [85]:
selected_cols = [
    "Hours_Studied",
    "Previous_Scores",
    "Tutoring_Sessions",
    "Exam_Score"
]

In [86]:
quartiles = df[selected_cols].quantile([0.25, 0.50, 0.75]).T

quartiles.columns = ["Q1", "Q2_median", "Q3"]

quartiles["IQR"] = quartiles["Q3"] - quartiles["Q1"]

quartiles

,Q1,Q2_median,Q3,IQR
Hours_Studied,16.0,20.0,24.0,8.0
Previous_Scores,63.0,75.0,88.0,25.0
Tutoring_Sessions,1.0,1.0,2.0,1.0
Exam_Score,65.0,67.0,69.0,4.0


In [87]:
student_index = 1000

student_score = df.loc[student_index, "Exam_Score"]

student_score

np.int64(75)

In [88]:
percentile = (df["Exam_Score"] < student_score).mean() * 100

print(
    f"The student scored {student_score}. "
    f"This means the student scored higher than approximately {percentile:.2f}% "
    "of the students in the dataset."
)

The student scored 75. This means the student scored higher than approximately 98.06% of the students in the dataset.


## What Are Outliers?

Outliers are values that are very different from most of the other values in a variable.

They can happen for different reasons:

- a real but unusual case
- a data entry error
- a measurement error
- a special situation in the dataset

For example, if most students studied between 16 and 24 hours, a student who studied 44 hours may be considered unusual.

Outliers are important because they can affect statistical measures such as the mean and standard deviation. They can also influence machine learning models.

However, outliers should not be removed automatically. First, we need to investigate whether they are valid observations or data quality problems.

### Checking Outliers

Now we will check possible outliers using the IQR method.

The idea is to calculate:

IQR = Q3 - Q1

Then we define two limits:

Lower Limit = Q1 - 1.5 * IQR

Upper Limit = Q3 + 1.5 * IQR

Values below the lower limit or above the upper limit are considered possible outliers.

This method is the same logic used in a boxplot, but here we will only calculate the information without creating a graph.

In [89]:
outlier_results = []

for col in selected_cols:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1

    lower_limit = Q1 - 1.5 * IQR
    upper_limit = Q3 + 1.5 * IQR

    below = (df[col] < lower_limit).sum()
    above = (df[col] > upper_limit).sum()
    total = below + above

    outlier_results.append([
        col,
        Q1,
        Q3,
        IQR,
        lower_limit,
        upper_limit,
        below,
        above,
        total
    ])

outlier_summary = pd.DataFrame(
    outlier_results,
    columns=[
        "Variable",
        "Q1",
        "Q3",
        "IQR",
        "Lower Limit",
        "Upper Limit",
        "Below Limit",
        "Above Limit",
        "Total Outliers"
    ]
)

outlier_summary

,Variable,Q1,Q3,IQR,Lower Limit,Upper Limit,Below Limit,Above Limit,Total Outliers
0,Hours_Studied,16.0,24.0,8.0,4.0,36.0,19,21,40
1,Previous_Scores,63.0,88.0,25.0,25.5,125.5,0,0,0
2,Tutoring_Sessions,1.0,2.0,1.0,-0.5,3.5,0,423,423
3,Exam_Score,65.0,69.0,4.0,59.0,75.0,27,76,103


After counting the outliers, we can inspect the actual values.

This helps us understand whether the outliers are extreme values, possible errors, or valid observations.

### Other Ways to Calculate Outliers

The IQR method is based on quartiles, but there are other ways to identify values that are very different from the rest of the data.

One common alternative is the Z-score method.

The Z-score measures how many standard deviations a value is from the mean.

Formula:

z = (x - mean) / standard deviation

Where:

- x is the value
- mean is the average of the variable
- standard deviation measures how spread out the values are

A common rule is:

- if z > 3, the value may be an unusually high outlier
- if z < -3, the value may be an unusually low outlier

This method works better when the variable has a distribution close to normal.

In [90]:
zscore_results = []

for col in selected_cols:
    mean = df[col].mean()
    std = df[col].std()

    z_scores = (df[col] - mean) / std

    outliers = df[(z_scores < -3) | (z_scores > 3)]

    zscore_results.append([
        col,
        mean,
        std,
        outliers.shape[0]
    ])

zscore_summary = pd.DataFrame(
    zscore_results,
    columns=["Variable", "Mean", "Standard Deviation", "Total Outliers"]
)

zscore_summary

,Variable,Mean,Standard Deviation,Total Outliers
0,Hours_Studied,19.977109,5.985460,22
1,Previous_Scores,75.066165,14.400389,0
2,Tutoring_Sessions,1.495296,1.233984,26
3,Exam_Score,67.252117,3.914217,52


The IQR method uses quartiles and is less affected by extreme values. It is usually safer when the data is not normally distributed.

The Z-score method uses the mean and standard deviation. It is useful when the data is approximately normal, but it can be more sensitive to extreme values because both the mean and standard deviation are affected by outliers.

In practice, it is common to compare both methods instead of relying on only one.

### Business Rule Outliers

Sometimes, outliers should be detected using domain rules instead of statistical formulas.

For example, if `Exam_Score` should range from 0 to 100, then any value above 100 is invalid, even if it is not considered an outlier by a statistical method.

In this case, the value 101 should be checked because it may violate the expected rule for exam scores.

In [91]:
df.loc[df["Exam_Score"] > 100, "Exam_Score"] = 100

## Correlation

Correlation measures the relationship between two numerical variables.

It helps us understand whether variables tend to increase or decrease together.


The correlation value ranges from -1 to 1:

- close to 1: strong positive relationship
- close to -1: strong negative relationship
- close to 0: weak or no linear relationship

Correlation does not mean causation. Even if two variables are correlated, it does not mean that one causes the other.

### Pearson and Spearman Correlation

Pearson and Spearman are two ways to measure correlation between variables.

Pearson measures a linear relationship.

- It works well when the variables change together in something close to a straight-line pattern.

- Example: as `Hours_Studied` increases, `Exam_Score` also increases in a mostly linear way.

Spearman measures a rank-based relationship.

- It checks whether higher values of one variable are generally associated with higher values of another variable, even if the relationship is not a straight line.

- Spearman is useful when the data has outliers or when the relationship is not linear.

In short:

Pearson = linear relationship  
Spearman = ranked or monotonic relationship

In [92]:
correlation_matrix = df.corr(numeric_only=True, method="pearson")

correlation_matrix

,Hours_Studied,Attendance,Sleep_Hours,Previous_Scores,Tutoring_Sessions,Physical_Activity,Exam_Score
Hours_Studied,1.000000,-0.004997,0.013119,0.022492,-0.009654,0.003183,0.445209
Attendance,-0.004997,1.000000,-0.019602,-0.018083,0.012070,-0.022976,0.580394
Sleep_Hours,0.013119,-0.019602,1.000000,-0.023238,-0.011466,-0.002898,-0.017149
Previous_Scores,0.022492,-0.018083,-0.023238,1.000000,-0.013158,-0.006694,0.174292
Tutoring_Sessions,-0.009654,0.012070,-0.011466,-0.013158,1.000000,0.016325,0.156769
Physical_Activity,0.003183,-0.022976,-0.002898,-0.006694,0.016325,1.000000,0.025155
Exam_Score,0.445209,0.580394,-0.017149,0.174292,0.156769,0.025155,1.000000


In [93]:
correlation_matrix = df.corr(numeric_only=True, method="spearman")

correlation_matrix

,Hours_Studied,Attendance,Sleep_Hours,Previous_Scores,Tutoring_Sessions,Physical_Activity,Exam_Score
Hours_Studied,1.000000,-0.005054,0.013260,0.022563,-0.009805,-0.003413,0.482343
Attendance,-0.005054,1.000000,-0.016212,-0.017939,0.012652,-0.024908,0.674109
Sleep_Hours,0.013260,-0.016212,1.000000,-0.023196,-0.004422,-0.001957,-0.006526
Previous_Scores,0.022563,-0.017939,-0.023196,1.000000,-0.018697,-0.003696,0.191116
Tutoring_Sessions,-0.009805,0.012652,-0.004422,-0.018697,1.000000,0.006986,0.164294
Physical_Activity,-0.003413,-0.024908,-0.001957,-0.003696,0.006986,1.000000,0.026179
Exam_Score,0.482343,0.674109,-0.006526,0.191116,0.164294,0.026179,1.000000


### Interpreting Correlation Strength

There is no perfect universal rule, but a common interpretation is:

- 0.00 to 0.19: very weak correlation
- 0.20 to 0.39: weak correlation
- 0.40 to 0.59: moderate correlation
- 0.60 to 0.79: strong correlation
- 0.80 to 1.00: very strong correlation

The same idea applies to negative values.

In [94]:
exam_correlation = correlation_matrix["Exam_Score"].drop("Exam_Score").sort_values(ascending=False)

corr_comments = []

for variable, corr in exam_correlation.items():
    abs_corr = abs(corr)

    if abs_corr < 0.20:
        strength = "very weak"
    elif abs_corr < 0.40:
        strength = "weak"
    elif abs_corr < 0.60:
        strength = "moderate"
    elif abs_corr < 0.80:
        strength = "strong"
    else:
        strength = "very strong"

    if corr > 0:
        direction = "positive"
    elif corr < 0:
        direction = "negative"
    else:
        direction = "no"

    corr_comments.append([variable, corr, direction, strength])

corr_comments = pd.DataFrame(
    corr_comments,
    columns=["Variable", "Correlation with Exam Score", "Direction", "Strength"]
)

corr_comments

,Variable,Correlation with Exam Score,Direction,Strength
0,Attendance,0.674109,positive,strong
1,Hours_Studied,0.482343,positive,moderate
2,Previous_Scores,0.191116,positive,very weak
3,Tutoring_Sessions,0.164294,positive,very weak
4,Physical_Activity,0.026179,positive,very weak
5,Sleep_Hours,-0.006526,negative,very weak


- The variable that stands out the most is `Attendance`. It has the strongest relationship with `Exam_Score`, with a correlation of 0.67. This suggests that attendance is much more associated with exam performance than the other variables.

- `Hours_Studied` also has a relevant relationship with `Exam_Score`, but it is weaker than attendance. This is interesting because study time matters, but it does not appear to explain exam performance as strongly as class attendance.

- `Previous_Scores` has a surprisingly low correlation with `Exam_Score`. We might expect previous academic performance to be strongly related to the final exam score, but in this dataset the relationship is very weak.

- `Tutoring_Sessions` also shows a very weak correlation. This may indicate that simply having more tutoring sessions is not strongly associated with higher exam scores, at least in a linear way.

- `Physical_Activity` and `Sleep_Hours` have almost no correlation with `Exam_Score`. In this dataset, they do not seem to have a meaningful relationship with exam performance.

## Hypothesis Testing

### Hypothesis Test 1: Average Attendance

Scenario:

The school claims that the average student attendance is higher than 80%.

We want to test whether the data provides statistical evidence to support this claim.

H0: The average Attendance is less than or equal to 80.

H1: The average Attendance is higher than 80.

Test:

One-sample t-test, right-tailed.

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_1samp.html

In [95]:
from scipy import stats

In [117]:
reference_value = 80

t_stat, p_value_two_tailed = stats.ttest_1samp(
    df["Attendance"],
    reference_value,
    alternative="greater"
)

if t_stat > 0:
    p_value = p_value_two_tailed / 2
else:
    p_value = 1 - (p_value_two_tailed / 2)

print("T statistic:", t_stat)
print("P-value:", p_value)

T statistic: 0.14417840621709863
P-value: 0.2213410370103028


In [118]:
alpha = 0.05

if p_value < alpha:
    print("Reject H0.")
    print("There is statistical evidence that the average Attendance is higher than 80%.")
else:
    print("Do not reject H0.")
    print("There is not enough evidence that the average Attendance is higher than 80%.")

Do not reject H0.
There is not enough evidence that the average Attendance is higher than 80%.


### Hypothesis Test 2: Attendance and Exam Score

Scenario:

Do students with higher attendance have higher exam scores?

We want to test whether there is a statistically significant positive relationship between `Attendance` and `Exam_Score`.

H0: There is no positive linear relationship between Attendance and Exam_Score.

H1: There is a positive linear relationship between Attendance and Exam_Score.

Test:

Pearson correlation test, right-tailed.

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.pearsonr.html

In [119]:
corr, p_value_two_tailed = stats.pearsonr(
    df["Attendance"],
    df["Exam_Score"],
    alternative="greater"
)

if corr > 0:
    p_value = p_value_two_tailed / 2
else:
    p_value = 1 - (p_value_two_tailed / 2)

print("Pearson correlation:", corr)
print("P-value:", p_value)

Pearson correlation: 0.5803937215306656
P-value: 0.0


In [120]:
alpha = 0.05

if p_value < alpha:
    print("Reject H0.")
    print("There is statistical evidence of a positive relationship between Attendance and Exam_Score.")
else:
    print("Do not reject H0.")
    print("There is not enough evidence of a positive relationship between Attendance and Exam_Score.")

Reject H0.
There is statistical evidence of a positive relationship between Attendance and Exam_Score.


## Standardization and Normalization

Some numerical variables can have very different scales.

For example:

- `Attendance` goes from 60 to 100
- `Hours_Studied` goes from 1 to 44
- `Exam_Score` goes from 55 to 101

Many machine learning models are sensitive to scale. When variables have very different ranges, some variables may influence the model more simply because their numbers are larger.

Two common ways to rescale numerical variables are:

Standardization

Normalization

### Standardization

Standardization transforms a variable so that it has:

- mean equal to 0
- standard deviation equal to 1

Formula:

standardized value = (value - mean) / standard deviation

In [110]:
from sklearn.preprocessing import StandardScaler

numeric_cols = [
    "Hours_Studied",
    "Attendance",
    "Sleep_Hours",
    "Previous_Scores",
    "Tutoring_Sessions",
    "Physical_Activity",
    "Exam_Score"
]

scaler = StandardScaler()

df_standardized = df.copy()

df_standardized[numeric_cols] = scaler.fit_transform(df[numeric_cols])

In [112]:
df_standardized[numeric_cols].describe().T

,count,mean,std,min,25%,50%,75%,max
Hours_Studied,6378.0,-1.693360e-16,1.000078,-3.170784,-0.664514,0.003825,0.672163,4.013856
Attendance,6378.0,5.458858e-16,1.000078,-1.733435,-0.867620,-0.001805,0.864009,1.729824
Sleep_Hours,6378.0,-1.030499e-16,1.000078,-2.067530,-0.705056,-0.023819,0.657418,2.019892
Previous_Scores,6378.0,4.634459e-16,1.000078,-1.740795,-0.837971,-0.004595,0.898229,1.731605
Tutoring_Sessions,6378.0,1.002647e-17,1.000078,-1.211858,-0.401411,-0.401411,0.409036,5.271718
Physical_Activity,6378.0,8.021179e-17,1.000078,-2.889375,-0.945447,0.026516,0.998480,2.942408
Exam_Score,6378.0,-2.027576e-16,1.000078,-3.131429,-0.575569,-0.064397,0.446775,8.369940


### Normalization

Normalization transforms a variable to a fixed range.

The most common range is from 0 to 1.

Formula:

normalized value = (value - minimum) / (maximum - minimum)

In [113]:
from sklearn.preprocessing import MinMaxScaler

normalizer = MinMaxScaler()

df_normalized = df.copy()

df_normalized[numeric_cols] = normalizer.fit_transform(df[numeric_cols])

df_normalized[numeric_cols].describe().T

,count,mean,std,min,25%,50%,75%,max
Hours_Studied,6378.0,0.441328,0.139197,0.0,0.348837,0.441860,0.534884,1.0
Attendance,6378.0,0.500521,0.288768,0.0,0.250000,0.500000,0.750000,1.0
Sleep_Hours,6378.0,0.505827,0.244672,0.0,0.333333,0.500000,0.666667,1.0
Previous_Scores,6378.0,0.501323,0.288008,0.0,0.260000,0.500000,0.760000,1.0
Tutoring_Sessions,6378.0,0.186912,0.154248,0.0,0.125000,0.125000,0.250000,1.0
Physical_Activity,6378.0,0.495453,0.171488,0.0,0.333333,0.500000,0.666667,1.0
Exam_Score,6378.0,0.272266,0.086953,0.0,0.222222,0.266667,0.311111,1.0
